### ISSUE TITLE
Fix abstraction failure in SiteUtil::getCheckMessageKeys

### ISSUE THREAD

**Issue by stoyanK7** [checkstyle/checkstyle#13500]
From https://github.com/checkstyle/checkstyle/pull/13423#discussion_r1280844252

We suppressed an `InstanceofChain` IDEA error in `SiteUtil::getCheckMessageKeys`. The violation comes from these 2 if-statements
```java
    if (module == RegexpMultilineCheck.class) {
        checkstyleMessageKeys.addAll(getCheckMessageKeys(
                Class.forName(
                        "com.puppycrawl.tools.checkstyle.checks.regexp.MultilineDetector"
                ), checkInstance)
        );
    }
    else if (module == RegexpSinglelineCheck.class
            || module == RegexpSinglelineJavaCheck.class) {
        checkstyleMessageKeys.addAll(getCheckMessageKeys(
                Class.forName(
                        "com.puppycrawl.tools.checkstyle.checks.regexp.SinglelineDetector"
                ), checkInstance)
        );
    }
```

We need to decide what to do with this.

> And this violation is right, it is hack.
We should have separate ticket to improve it by refactoring of modules to have all MSG fields in module as field, all child, nested, composited classes should take by constructor content of message.
We already discussed it multiple times, and now we see real need for this

**Comment by Carbon14-48** [comment]
Hi, I'd like to work on this issue. I'll move the message keys from the detector classes to the main checks and remove the hack from SiteUtil.

**Comment by romani** [comment]
fix is merged
